etl/tests/test_flask_app.py

```diff
@@ -34,6 +36,26 @@ def test_client(monkeypatch: _pytest.monkeypatch.MonkeyPatch, database_name: str
     return flask_app.App.test_client()
 
 
+@pytest.fixture()
+def thread_runner() -> typing.Generator:
+    yield flask_app.ThreadRunner
+    while flask_app.ThreadRunner.is_thread_running():
+        time.sleep(0.1)
+
+
+def test_threadrunner(thread_runner: flask_app.ThreadRunner) -> None:
```

Comment:
This seems like a flakey test.
Sleep is not guaranteed to block for exactly the time it is given, it is guaranteed to block for <i>at least</i> that long, and may run for an arbitrary amount longer due to scheduling of other activities.
I don't know of a great way around that, but maybe something like the `sleeper` function instead of just sleeping checking a `flag` variable that is scoped to `test_threadrunner` above it in a `while` loop, then switching that flag to turn it off?